### PR TITLE
NIFI-6919: Added relationship attribute to DistributeLoad

### PR DESCRIPTION
**Pull request by m-hogue**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

Adds a FlowFile attribute to the `DistributeLoad` processor as well as a feature flag controlling whether that attribute is written.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [x] Have you written or updated unit tests to verify your changes?
- [x] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
I'm wondering if it really needs a property... I would just add the attribute by default in my opinion, no?

**Comment by m-hogue** [comment]
> I'm wondering if it really needs a property... I would just add the attribute by default in my opinion, no?

That's okay with me! I wanted to make it as flexible as possible in case there was a scenario this attribute wasn't desired. I can drop the feature flag.

**Comment by m-hogue** [comment]
@pvillard31 : Change has been made. Let me know if there are any other changes necessary.

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DistributeLoad.java [hunk 1]
```diff
@@ -71,6 +73,9 @@
         + "'5' will be receive 10 FlowFiles in each iteration instead of 1.")
 @DynamicRelationship(name = "A number 1..<Number Of Relationships>", description = "FlowFiles are sent to this relationship per the "
         + "<Distribution Strategy>")
+@WritesAttributes(
+        @WritesAttribute(attribute = "distribute.load.relationship", description = "If enabled, the name of the specific relationship a flow file was routed through will be written to the FlowFile")
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/DistributeLoad.java]:
```suggestion
        @WritesAttribute(attribute = "distribute.load.relationship", description = "The name of the specific relationship the flow file has been routed through")
```

**Comment by mattyb149** [comment]
+1 LGTM, made Pierre's change to the attribute description. Thanks for the improvement! Merging to master

**Comment by m-hogue** [comment]
Thanks so much, @mattyb149!

**Pull request closed without merging**